cdk/lib/nishikiBackend-stack.ts

```diff
@@ -16,6 +19,74 @@ export class NishikiBackendStack extends cdk.Stack {
 	constructor(scope: Construct, id: string, props: IProps) {
 		super(scope, id, props);
 
-		const { stage, api, table, userPool } = props;
+		const { stage, table, userPool } = props;
+
+		const mainFunction = nishikiMainBackendFunction(this, stage, {
+			tableName: table.tableName,
+		});
+
+		// add permission to writing and reading of Dynamo DB.
+		table.grantReadWriteData(mainFunction);
+
+		// add API Gateway
+		const api = new apigateway.LambdaRestApi(this, "NishikiApi", {
+			restApiName: `nishiki-endpoint-${stage}-api-gateway`,
+			handler: mainFunction,
+			proxy: true,
+			defaultMethodOptions: {
+				authorizer: nishikiBackendAPIAuthorizer(this, userPool),
+			},
+			deployOptions: {
+				stageName: `${stage}`,
+			},
+		});
```

Comment:
How about using a function to create APIGateway as other resources do?
```suggestion
		// add API Gateway
		const api = nishikiApiGateway(this, stage, {
			mainFunction,
			userPool,
		});
```

```typescript
interface IApiGatewayProps {
	mainFunction: lambda.Function;
	userPool: UserPool;
}

const nishikiApiGateway = (
	scope: Construct,
	stage: Stage,
	props: IApiGatewayProps,
) => {
	const { mainFunction, userPool } = props;
	new apigateway.LambdaRestApi(scope, "NishikiApi", {
		restApiName: `nishiki-endpoint-${stage}-api-gateway`,
		handler: mainFunction,
		proxy: true,
		defaultMethodOptions: {
			authorizer: nishikiBackendAPIAuthorizer(scope, userPool),
		},
		deployOptions: {
			stageName: `${stage}`,
		},
	});
};
```

Reply:
I thought so, but in this case, I wanted to write it explicitly for the future. If you want it to be encapsulated, it should be needed in the path as an argument.  But by doing this, you can recognize this is immature. This is the reason I wrote this.

Of course, I understand your pointing it out, and I agree with your opinion.  I just want to explain why I wrote this. Do you think we should this in the function?

Reply:
> it should be needed in the path as an argument

Sorry, I didn't get this part. Could you elaborate it?

Reply:
I mean, the reason for creating a function is the possibility of using it again. To open to future updating, I thought it needed to take a path as an argument.

```ts
export class NishikiBackendStack extends cdk.Stack {
	constructor(scope: Construct, id: string, props: IProps) {
		super(scope, id, props);

		const { stage, table, userPool } = props;

		const mainFunction = nishikiMainBackendFunction(this, stage, {
			tableName: table.tableName,
		});

		// add permission to writing and reading of Dynamo DB.
		table.grantReadWriteData(mainFunction);

		const auth = nishikiBackendAPIAuthorizer(this, userPool);

		// add API Gateway
		const api = new apigateway.RestApi(this, "NishikiApi");
		
		addResource({
			api,
			runFunction: mainFunction,
			path: "main",
			userPoolAuthorizer: auth,
		})
	}
}

interface INewResource {
	api: apigateway.RestApi,
	runFunction: IFunction,
	path: string,
	userPoolAuthorizer: CognitoUserPoolsAuthorizer
}

const addResource = (props: INewResource) => {
	const { api, runFunction, path, userPoolAuthorizer } = props;

	api.root.addResource(path).addProxy({
		anyMethod: true,
		defaultIntegration: new apigateway.LambdaIntegration(runFunction),
	})

}

```
Not taking it this way, I thought I could show it could take only this way.

But, you know, this is my personal preference. Please tell me your opinion.

Reply:
Ohhh, I see. Thank you so much for your explanation. Considering the future system, the function should have Path as you said! So in this case, for now it's fine to create the APIGateway without a function cuz we have only one path and proxy now. But honestly I think if we ignore the possibility of the future change, it's also acceptable to make a function to create the API Gateway without the Path argument.

Reply:
Thank you for reviewing and I made sense your opinion. I think you are right. But to get the agility, I merge this PR. Let's check it after completing tasks related to CDK.